src/components/PlaidLink/index.js

```diff
@@ -1,44 +1,77 @@
-import {useCallback, useEffect} from 'react';
-import {usePlaidLink} from 'react-plaid-link';
+import React from 'react';
+import {View} from 'react-native';
+import styles from '../../styles/styles';
 import {plaidLinkPropTypes, plaidLinkDefaultProps} from './plaidLinkPropTypes';
-import Log from '../../libs/Log';
-
-const PlaidLink = (props) => {
-    const onSuccess = useCallback((publicToken, metadata) => {
-        props.onSuccess({publicToken, metadata});
-    }, []);
-
-    const {open, ready, error} = usePlaidLink({
-        token: props.token,
-        onSuccess,
-        onExit: (exitError, metadata) => {
-            Log.info('[PlaidLink] Exit: ', false, {exitError, metadata});
-            props.onExit();
-        },
-        onEvent: (event, metadata) => {
-            Log.info('[PlaidLink] Event: ', false, {event, metadata});
-        },
-
-        // The redirect URI with an OAuth state ID. Needed to re-initialize the PlaidLink after directing the
-        // user to their respective bank platform
-        receivedRedirectUri: props.receivedRedirectURI,
-    });
-
-    useEffect(() => {
-        if (error) {
-            props.onError(error);
-            return;
+
+class PlaidLink extends React.Component {
+    constructor(props) {
+        super(props);
+        this.onMessage = this.onMessage.bind(this);
+        this.state = {
+            isHidden: false,
+        };
+    }
+
+    componentDidMount() {
+        window.addEventListener('plaidEvent', this.onMessage);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('plaidEvent', this.onMessage);
+    }
+
+    onMessage(e) {
+        switch (e.detail.eventName) {
```

Comment:
Should there be an early return just in case `e.detail` is undefined? If it ever happens, it would cause a fatal error and halt program execution. Either an early return, or use `lodashGet()` to access the properties safely.